pilot/pkg/config/kube/ior/route.go

```diff
@@ -583,8 +585,13 @@ func (r *route) findService(gateway *networking.Gateway) (string, string, error)
 		gwSelector.String(), namespaces)
 }
 
-func getRouteName(namespace, name, actualHost string) string {
-	return fmt.Sprintf("%s-%s-%s", namespace, name, hostHash(actualHost))
+func getRouteName(namespace, name, actualHost, serviceNamespace string) string {
```

Comment:
Why add `serviceNamespace` here?

Reply:
I need to know what is the ingress controller namespace and its length because when the hostname is set to "*" we let [OpenShift creates the route](https://github.com/maistra/istio/blob/maistra-2.3/pilot/pkg/config/kube/ior/route.go#L603-L607). 
In this case, the Route `spec.host` is set following the format `gateway_namespace-gateway_name-ingressgateway_namespace` (`Gateway.metadata.name` + ingress controller namespace) and it shouldn't exceed 63 characters.